Tests/StoreKitUnitTests/StoreKit2StorefrontListenerTests.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  StoreKit2StorefrontListenerTests.swift
+//
+//  Created by Juanpe Catalán on 9/5/22.
+
+import Nimble
+@testable import RevenueCat
+import XCTest
+
+@available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+class StoreKit2StorefrontListenerTests: XCTestCase {
+
+    private var listener: StoreKit2StorefrontListener! = nil
+
+    override func setUp() {
+        super.setUp()
+
+        self.listener = .init(delegate: nil)
+    }
+
+    func testStopsListeningToChanges() throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        var handle: Task<Void, Never>?
+
+        expect(self.listener!.taskHandle).to(beNil())
```

Comment:
You don't need the `!`. It's better if tests assume it's non-nil :)
```suggestion
        expect(self.listener.taskHandle).to(beNil())
```